services/ux-backend/handlers/onboarding/clienttokens/handler.go

```diff
@@ -59,7 +60,13 @@ func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int
 		storageQuotaInGiB = ptr.To(unitAsGiB * quota.Value)
 	}
 
-	if onboardingToken, err := util.GenerateClientOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath, storageQuotaInGiB); err != nil {
+	storageCluster, err := util.GetStorageClusterInNamespace(r.Context(), cl, namespace)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
This cannot be an `http.StatusBadRequest` because the failure has nothing to do with the content of the request, which also means that the user cannot change the request and clear the check. 
For these kinds of issues, I would suggest using `http.StatusInternalServerError`